src/mlpack/core/data/string_encoding.hpp

```diff
@@ -122,9 +123,11 @@ class StringEncoding
   *
   */
   template<typename TokenizerType>
-  void Encode(const std::vector<std::string>& input,
-              std::vector<std::vector<size_t>>& output,
-              TokenizerType tokenizer);
+  typename std::enable_if<std::is_same< EncodingPolicy,
+      data::DictionaryEncoding>::value, void>::type Encode(const
+      std::vector<std::string>& input,
+      std::vector<std::vector<size_t>>& output,
+      TokenizerType tokenizer);
```

Comment:
I suggest to move `enable_if` to the list of arguments.

Reply:
Probably it's a good idea to introduce some kind of traits instead of `std::is_same`.